annotation-processing/src/main/java/processor/FastProcessor.java

```diff
@@ -0,0 +1,96 @@
+package processor;
+
+import com.google.auto.service.AutoService;
+
+import javax.annotation.processing.*;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.tools.Diagnostic;
+import javax.tools.JavaFileObject;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+@SupportedAnnotationTypes("processor.FastType")
+@SupportedSourceVersion(SourceVersion.RELEASE_8)
+@AutoService(Processor.class)
+public class FastProcessor extends AbstractProcessor {
+    @Override
+    public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+        Map<String, Element> nameClassMap = processRound(annotations, roundEnv);
+
+        if(!nameClassMap.isEmpty()){
+            try {
+                buildCreatorClass(nameClassMap);
+            } catch (IOException e) {
+                processingEnv.getMessager().printMessage(Diagnostic.Kind.ERROR, e.getMessage());
+            }
+        }
+        return true;
+    }
+
+    private Map<String, Element> filterDuplicates(Map<String, Set<Element>> nameClassMap) {
+        Map<String, Element> finalNameClassMap = new HashMap<>();
+        for (Map.Entry<String, Set<Element>> entry : nameClassMap.entrySet()) {
+            String fastTypeName = entry.getKey();
+            Set<Element> elements = entry.getValue();
+            if (elements.size() > 1) {
+                reportDuplicate(fastTypeName, elements);
+            }else {
+                finalNameClassMap.put(fastTypeName, elements.stream().findFirst().get());
+            }
+        }
+        return finalNameClassMap;
+    }
+
+    private Map<String, Element> processRound(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+        Map<String, Set<Element>> currentNameClassMap = new HashMap<>();
```

Comment:
rename to `nameClassMap`, change variable type to HashMap. There is no need to work through interfaces here. Access via interface is usually a bit slower

Reply:
The same for Set. It's our internal thing, we don't need to hide implementation details and return generic Set interface